src/modules/downloader.js

```diff
@@ -166,39 +166,81 @@ module.exports.downloadTemplate = (template) => {
 module.exports.downloadAndUpdateBinaries = async (latest = false) => {
     await downloadBinariesFromRelease(latest);
     utils.log('Finalizing and cleaning temp. files.');
-    if(!fse.existsSync('bin'))
+    if (!fse.existsSync('bin'))
         fse.mkdirSync('bin');
 
-    for(let platform in constants.files.binaries) {
-        for(let arch in constants.files.binaries[platform]) {
+    for (let platform in constants.files.binaries) {
+        for (let arch in constants.files.binaries[platform]) {
             let binaryFile = constants.files.binaries[platform][arch];
-            if(fse.existsSync(`.tmp/${binaryFile}`)) {
+            if (fse.existsSync(`.tmp/${binaryFile}`)) {
                 fse.copySync(`.tmp/${binaryFile}`, `bin/${binaryFile}`);
             }
         }
     }
 
-    for(let dependency of constants.files.dependencies) {
-        fse.copySync(`.tmp/${dependency}`,`bin/${dependency}`);
+    for (let dependency of constants.files.dependencies) {
+        fse.copySync(`.tmp/${dependency}`, `bin/${dependency}`);
     }
     utils.clearDirectory('.tmp');
 }
 
 module.exports.downloadAndUpdateClient = async (latest = false) => {
     const configObj = config.get();
-    if(!configObj.cli.clientLibrary) {
+    if (!configObj.cli.clientLibrary) {
         utils.log(`neu CLI won't download the client library --` +
-                    ` download @neutralinojs/lib from your Node package manager.`);
+            ` download @neutralinojs/lib from your Node package manager.`);
         return;
     }
     const clientLibrary = utils.trimPath(configObj.cli.clientLibrary);
     await downloadClientFromRelease(latest);
     await downloadTypesFromRelease(latest);
     utils.log('Finalizing and cleaning temp. files...');
     fse.copySync(`.tmp/${constants.files.clientLibraryPrefix + getScriptExtension()}`
-            , `./${clientLibrary}`);
+        , `./${clientLibrary}`);
     fse.copySync(`.tmp/neutralino.d.ts`
-            , `./${clientLibrary.replace(/[.][a-z]*$/, '.d.ts')}`);
+        , `./${clientLibrary.replace(/[.][a-z]*$/, '.d.ts')}`);
     utils.clearDirectory('.tmp');
 }
 
+module.exports.checkIfTemplateValid = (template) => {
+    return new Promise((resolve) => {
+        https.get(constants.remote.templateCheckUrl.replace('{template}', template), {
+            headers: {
+                'X-GitHub-Api-Version': '2022-11-28',
+                "Accept": "application/vnd.github+json",
```

Comment:
Do we need to set these two HTTP headers? Thanks :tada:

Reply:
The `'Accept': 'application/vnd.github+json'` header specifies that we want the response in JSON format. However, since JSON is the default format, we can ignore this header.

The `'X-GitHub-Api-Version': '2022-11-28'` header specifies the API version we want to use. If we don't specify this, it will default to the initial release API version, which is also `'2022-11-28'`, so we can remove that header as well.

ps: raised a pr for this 🎉